platform/api/brightbox/api.go

```diff
@@ -0,0 +1,259 @@
+			return fmt.Errorf("unmaping cloud IP from API: %w", err)
+		}
+		plog.Info("Cloud IP released.")
+	}
+
+	if _, err := a.client.DestroyServer(context.TODO(), id); err != nil {
+		return fmt.Errorf("destroying server from API: %w", err)
+	}
+
+	return nil
+}
+
+func (a *API) GC(gracePeriod time.Duration) error {
+	threshold := time.Now().Add(-gracePeriod)
+	// TODO: CloudIP has no creation date for now.
+	// We can't safely delete "old" cloud IPs.
+	// NOTE: Currently, cloud IPs removal is implemented as an independant
+	// 'ore' subcommand.
+
+	servers, err := a.client.Servers(context.TODO())
+	if err != nil {
+		return fmt.Errorf("listing servers from API: %w", err)
+	}
+
+	for _, server := range servers {
+		if server.Status == serverstatus.Deleted || server.CreatedAt.After(threshold) {
+			continue
+		}
+
+		if err := a.DeleteServer(server.ID); err != nil {
+			return fmt.Errorf("deleting server: %w", err)
+		}
+	}
+
+	images, err := a.client.Images(context.TODO())
+	if err != nil {
+		return fmt.Errorf("listing servers from API: %w", err)
+	}
+
+	for _, image := range images {
+		if image.Public || image.Status == imagestatus.Deleted || image.CreatedAt.After(threshold) {
+			continue
+		}
+
+		if err := a.DeleteImage(image.ID); err != nil {
+			return fmt.Errorf("deleting image: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (a *API) GetConsoleOutput(id string) (string, error) {
+	// NOTE: There is no way to get console output from the API.
+	// A workaround would be the fetch the console_url + console_token to read from this
+	// endpoint.
+	return "", nil
+}
+
+// UploadImage will upload an image from the URL on Brightbox and wait for it to become
+// available.
+func (a *API) UploadImage(name, URL string) (string, error) {
+	defaultUsername := "core"
+
+	img, err := a.client.CreateImage(context.TODO(), brightbox.ImageOptions{
+		Name:     &name,
+		URL:      URL,
+		Username: &defaultUsername,
+		Arch:     arch.X86_64,
+	})
+	if err != nil {
+		return "", fmt.Errorf("creating image from API: %w", err)
+	}
+
+	// It usually takes around 20 seconds to extract the image.
+	if err := util.WaitUntilReady(2*time.Minute, 5*time.Second, func() (bool, error) {
+		image, err := a.client.Image(context.TODO(), img.ID)
+		if err != nil {
+			return false, fmt.Errorf("getting image status: %w", err)
+		}
+
+		return image.Status == imagestatus.Available, nil
+	}); err != nil {
+		a.DeleteImage(img.ID)
+		return "", fmt.Errorf("getting image active: %w", err)
+	}
+
+	return img.ID, nil
+}
+
+// RemoveCloudIPs remove any left overs IPs.
+func (a *API) RemoveCloudIPs() error {
+	cloudIPs, err := a.client.CloudIPs(context.TODO())
+	if err != nil {
+		return fmt.Errorf("getting cloud IPs: %w", err)
+	}
+
+	for _, cloudIP := range cloudIPs {
+		// Do not remove a mapped cloud IP - otherwise, we can end up
+		// with a server without public IP.
+		if cloudIP.Status == cloudipstatus.Mapped {
```

Comment:
I thought that makes sense to ensure that the kola run can use all available IP addrs. I guess we would need a "delete-vms" ore command to also run before this here in the Jenkins job.

Reply:
Because the garbage collection comes too late when a job restarts after kola crashed.